govuk_chat_evaluation/rag_answers/custom_deepeval/metrics/coherence/template.py

```diff
@@ -0,0 +1,98 @@
+    "STEP 6 - Logical Flow Verification: Check if information builds logically or if the user must mentally reorder it. Ask: 'Does each sentence/paragraph follow naturally from the previous one?' and 'Would moving any sentence earlier make the answer clearer?' If critical information appears AFTER less important details, or if the answer jumps between topics without transitions, consider scores 2-4 depending on severity.",
+    "STEP 7 - Practical Utility Test: After reading the answer, ask: 'Could the user confidently explain back: (a) what applies to their situation, (b) what they need to do next, (c) what's in their control vs others' control?' If you cannot answer all three clearly, the answer has failed coherence - consider scores 1-3.",
+    "STEP 8 - Apply the Harshness Principle: LLM-generated text often APPEARS coherent because it flows smoothly. You must look BEYOND surface readability. Ask: 'Even though this reads nicely, could a stressed/distracted/average-literacy user misunderstand what they should do?' If yes, do not give a 5. Remember: Score 1-2 is for answers that logically fail the user, not just answers that are grammatically poor. A well-written answer with subtle contradictions deserves a 1.",
+    "STEP 9 - Final Coherence-Only Check: Confirm your evaluation focuses solely on: structure, clarity, consistency, perspective, and logical presentation. You are NOT evaluating: correctness, completeness, relevance, or politeness. An answer can say 'I don't know' and score a 5 if it's clear about what it does/doesn't know. An answer can be comprehensive and score a 1 if it contradicts itself.",
+    "STEP 10 - Score Assignment with Justification: Based on Steps 1-9, assign a score using this logic: (a) ANY contradiction on core actions/obligations = 1 or 2, (b) Wrong/mixed perspective or buried critical context = 2 or 3, (c) Notable ambiguity or flow issues = 3 or 4, (d) Minor inefficiency only = 4, (e) No issues detected = 5. Provide specific evidence from the answer for your score. Default to the LOWER score if uncertain between two.",
+]
+
+
+COHERENCE_EVALUATION_TEMPLATE = """You are evaluating the coherence of a government service response. Given the evaluation steps and rubric below, assess how well-structured and logically connected the response is.
+
+Return a JSON object with:
+- `"score"`: a number between {lo} and {hi}, based on the rubric
+- `"reason"`: a focused explanation of the coherence assessment, referencing specific structural elements and logical flow
+
+Only return valid JSON, no other text.
+
+---
+
+Evaluation Steps:
+{evaluation_steps}
+
+{rubric_text}
```

Comment:
Similarly seems like the rubric information can be hardcoded here to make things simpler - again unless I'm missing an intended function?